frontend/kendra-button-front/src/components/Authenticator.tsx

```diff
@@ -1,57 +1,119 @@
-import { AmplifyAuthenticator, AmplifySignUp } from '@aws-amplify/ui-react';
-import { ReactElement, ReactNode, useEffect } from 'react';
+import {
+  Dispatch,
+  ReactElement,
+  ReactNode,
+  SetStateAction,
+  useEffect,
+  useState,
+} from 'react';
 
-// https://github.com/aws-amplify/amplify-js/issues/5825#issuecomment-631759616
-import { onAuthUIStateChange } from '@aws-amplify/ui-components';
+import { Auth, Hub } from 'aws-amplify';
+import { AuthState } from '@aws-amplify/ui-components';
+
+import { SignUp } from '../components';
 
 interface Props {
+  setUser: Dispatch<SetStateAction<any>>;
   children: ReactNode;
-  onStateChange?: any;
   isLoggedIn: boolean;
 }
 const Authenticator = (props: Props): ReactElement => {
-  const { children, onStateChange, isLoggedIn } = props;
+  const { children, setUser } = props;
+  const [screen, setScreen] = useState(AuthState.SignIn);
+
+  const checkUser = async (retry, tryCnt = 1): Promise<void> => {
+    const tryLimit = 3;
+    console.log('[checkUser retryflag]', retry);
+    try {
+      const user = await Auth.currentAuthenticatedUser();
+      console.log('[user:Auth]', user);
+      setScreen(AuthState.SignedIn);
+      setUser(user);
+    } catch (e) {
+      console.log('[error in checkUser]', e);
+      if (retry === true && tryLimit > tryCnt) {
+        setTimeout(() => {
+          checkUser(retry, tryCnt + 1);
+        }, 1000);
+      }
+    }
+  };
 
   useEffect(() => {
-    return onAuthUIStateChange((newAuthState, user) => {
-      onStateChange(newAuthState, user);
+    checkUser(true);
```

Comment:
파라미터 내용을 false로 변경 부탁드립니다!
true로 하면 1초간격으로 3회 유저 정보를 확인하는데,
그건 현재 Hub가 하고 있는 일을 똑같이 하는 꼴이 되어서요 ㅎㅎ
```suggestion
    checkUser(false);
```